lib/iris/experimental/um.py

```diff
@@ -520,7 +526,14 @@ def constants(name, dtype):
                     else:
                         offset = running_offset
                         data_provider = data_class(source, offset, word_size)
-                    klass = _FIELD_CLASSES[raw_headers[Field.LBREL_OFFSET]]
+                    lbrel_value = raw_headers[Field.LBREL_OFFSET]
+                    if lbrel_value == mdi_value:
```

Comment:
(as previously discussed)
I think that it probably desirable to use the proper MDI value from the integer_constants section here (that is ```self.integer_constants[21]```).
Despite what I said before, from inspecting the PP code, I *don't* think we need to handle either the stated MDI _or_ the default value.